include/invert_quda.h

```diff
@@ -771,21 +783,29 @@ namespace quda {
     const DiracMatrix &matSloppy;
     bool init;
 
-    Complex *W; // inner product matrix
-    Complex *C; // inner product matrix
-    Complex *alpha;
-    Complex *beta;
-    Complex *phi;
+    bool lambda_init;
+    QudaCABasis basis;
+
+    Complex *Q_AQandg; // Fused inner product matrix
+    Complex *Q_AS; // inner product matrix
+    Complex *alpha; // QAQ^{-1} g
+    Complex *beta; // QAQ^{-1} QpolyS
 
     ColorSpinorField *rp;
     ColorSpinorField *tmpp;
     ColorSpinorField *tmpp2;
     ColorSpinorField *tmp_sloppy;
     ColorSpinorField *tmp_sloppy2;
 
-    std::vector<ColorSpinorField*> r;  // residual vectors
-    std::vector<ColorSpinorField*> q;  // mat * residual vectors
-    std::vector<ColorSpinorField*> p;  // CG direction vectors
+    std::vector<ColorSpinorField*> S;  // residual vectors
+    std::vector<ColorSpinorField*> AS; // mat * residual vectors. Can be replaced by a single temporary.
+    std::vector<ColorSpinorField*> Q;  // CG direction vectors
```

Comment:
clang-format suggested the following change:


```
-    std::vector<ColorSpinorField*> Q;  // CG direction vectors
+    std::vector<ColorSpinorField*> Q; // CG direction vectors
```

<!-- identifier: clang-format -->
<!-- File: include/invert_quda.h -->